src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets

```diff
@@ -491,14 +492,17 @@ The .NET Foundation licenses this file to you under the MIT license.
         strip -no_code_signature_warning $(_StripFlag) &quot;$(NativeBinary)&quot;" />
   </Target>
 
-  <!-- NativeAOT-LLVM: separate target to reduce conflicts -->
-  <Target Name="LinkNativeLlvm"
-      Inputs="@(NativeObjects);@(NativeLibrary)"
-      Outputs="$(NativeBinary)"
-      DependsOnTargets="$(LinkNativeDependsOn)">
+  <Import Project="$(MSBuildThisFileDirectory)DotNetJsApi.targets" Condition="'$(_targetArchitecture)' == 'wasm' and '$(DotNetJsApi)' == 'true'" />
 
+  <Target Name="SetupLlvmLinker">
     <MakeDir Directories="$([System.IO.Path]::GetDirectoryName($(NativeBinary)))" />
 
+    <PropertyGroup>
+      <EmccStackSize Condition="'$(EmccStackSize)' == ''">5MB</EmccStackSize>
```

Comment:
What is the difference between `-s STACK_SIZE` and `-s TOTAL_STACK`?

Reply:
```
3.1.25 - 11/08/22
-----------------
- The `TOTAL_STACK` setting was renamed to `STACK_SIZE`.  The old name will
  continue to work as an alias. (#18128)
```

Reply:
I think Mono needs large stack because when we tier up some of the nested generics of LINQ, the stack goes really deep. Hope I remember the right reason. Does NativeAOT run the unit test suite yet ?

Reply:
> Does NativeAOT run the unit test suite yet ?

Not really, it is being worked on though (#2455).

1MB is the stack size on Windows, where regular NAOT tests run (and succeed without SO). WASM stack usage in our codegen strategy will be as follows:
1) Native (LLVM) stack usage: _less_ than native because:
   - Some data that would usually be spilled is put into WASM locals instead.
   - Some data would go on the "shadow stack". These are things with GC pointers in them, as well as EH-live variables.
      - This "shadow stack" is basically the same as the native LLVM stack, but controlled by us. It's size is currently not configurable at 1MB as well.

So, from that I am pretty sure 1MB is more than necessary at this point.

Reply:
Ook, droped 5MB and pass `EmccStackSize` to `IlcWasmStackSize` if set by user